fastlane_core/lib/fastlane_core/helper.rb

```diff
@@ -165,13 +165,17 @@ def self.keychain_path(name)
       # has changed for some users: https://github.com/fastlane/fastlane/issues/5649
       #
 
+      # Remove the ".keychain" at the end of the name
+      name.sub!(/\.keychain$/, "")
+
       keychain_paths = [
         File.join(Dir.home, 'Library', 'Keychains', name),
-        File.join(Dir.home, 'Library', 'Keychains', "#{name}-db"),
-        name,
-        "#{name}-db"
+        name
       ].map { |path| File.expand_path(path) }
 
+      # Transforms ["thing"] to ["thing", "thing-db", "thing.keychain", "thing-db.keychain"]
+      keychain_paths = keychain_paths.product(["", "-db"], ["", ".keychain"]).map(&:join)
```

Comment:
IMO adding/changing one line in the `keychain_paths` is less error prone than modifying multiple lines (in the future, not in this PR), and we have the comment there explaining what happens. Otherwise, something like this will happen inevitably which is hard to spot and will only lead to more headaches:

```ruby
keychain_paths = [
         File.join(Dir.home, 'Library', 'Keychains', name),
          #...
          name,
          "#{name}-db",
          "#{name}-db.keychain",
          "#{name}.keychain",
          File.join(Dir.home, 'Library', 'Keychains', new_option),
          File.join(Dir.home, 'Library', 'Keychains', "#{new_option}-db"),
          File.join(Dir.home, 'Library', 'Keychains', "#{new_option}.keychain"),
].map { |path| File.expand_path(path) }
```

Do you think we should expand more on the comment in L176 to make it more obvious what happens? Or should I just leave it like that?